### ISSUE TITLE
Test fails with version above 9.1.1

### ISSUE THREAD

**Issue by StickFigureJakob** [cypress-io/cypress#20504]
### Current behavior

testing old angularjs code fails after updating cypress from 9.1.1 to 9.3.1, 9.4.1 or 9.5.0.

Inside the app there is a test where changing the route state calls angular-ui-router.min.js, which then calls angular-animate.min.js. 

Some components of the application cannot be re-rendered, and it's not possible to interact with them anymore. Some actions like "change the tab" that would usually close the components did not do so, the broken parts stay on screen.

It works when doing it manuelly and don't fails with cypress:9.1.1.
It fails indipendent of the headless mode with electron and chrome.

## How it looks like
![9 5 0 Example](https://user-images.githubusercontent.com/85215602/156988198-829bfe00-d9dd-4fc4-8ab2-30abb7544663.png)

## Error message that appears only with cypress above version 9.1.1
![errorMessage](https://user-images.githubusercontent.com/85215602/156988406-b0d0efad-eaae-46d1-8701-aaec5b2795c2.png)

### Desired behavior

It should work, like it did with cypress 9.1.1

## How it should look
![9 1 1 Example](https://user-images.githubusercontent.com/85215602/156988228-2095cf27-cbb8-4aec-a604-f94d3d6f42cc.png)

### Test code to reproduce

[Repo with failing test setup](https://github.com/bettermarks/cypress-bug)

### Cypress Version

9.5.0

**Comment by StickFigureJakob** [comment]
![Bildschirmfoto vom 2023-05-24 14-35-59](https://github.com/cypress-io/cypress/assets/85215602/4ecfedc9-2d0f-44e2-9337-b07c6b47679b)

Still a problem with cypress 12

**Comment by warrensplayer** [comment]
@StickFigureJakob Are you able to create a simple reproduction that shows this error?  That would help us track down the issue the fastest.

**Comment by karfau** [comment]
@warrensplayer what exactly is not good enough regarding the existing reproduction?

The thing is that this is the legacy part of our application which we are in a process of getting rid of, so we try to spend as little time as possible in that code base, and it will most likely take us three more quarters to reach that goal of replacing it. I assume at that point cypress will be at version 13 or 14. While we are stuck at version 9.1.1

Something must have changed after version 9.1.1 with regards to that "runner" property that causes the runtime error.
Yes the frontend vode fir the reproction we provide is quite complex, but the issue doesn't occur in version 9.1.1

I'm not sure we would be able to provide a simpler repro, but if we try, we would need to undersrand how it is supposed to help.

**Comment by warrensplayer** [comment]
@karfau I totally missed the link to your repro project. I pulled that down, upgraded it to 12.14 this is how I got it to work.

First, in v12 and above, [testIsolation](https://docs.cypress.io/guides/core-concepts/test-isolation) is now the default behavior.  You will have to add `testIsolation: false` to the `e2e` section of your upgraded `cypress.config.js` file. 

For the problem, it looks like you are hitting a bug in the 1.4.7 version of `angular-animate`. This bug was fixed in `1.5.0-rc.0`.  See PR: https://github.com/angular/angular.js/pull/13424/files

I pulled in the source for angular-animate 1.4.7 and made the same fix to `src/ngAnimate/animateQueue.js` by wrapping the swtich statement with the if check on `animationDetails` and got it to work.  See video.

Good look with your project to replace your legacy code!

https://github.com/cypress-io/cypress/assets/1702361/9fb6cd33-734f-4f57-ba6a-f635c64646fc
